### ISSUE TITLE
have `windows_feature` install all dependencies by default

### ISSUE THREAD

**Issue by joshoohaah** [TAMUArch/cookbook.windows_ad#111]
the `default.rb` of this cookbook should by default have the dependencies installed for each feature.  see https://github.com/chef-cookbooks/windows#windows_feature  for explanation.
Currently the `ActiveDirectory-Powershell` feature fails out in the default.rb as other dependencies are not installed.

*current style*
```
    windows_feature feature do
      action :install
    end
```
*proposed new format*
```
    windows_feature feature do
      all true
      action :install
    end
```

**Comment by stuartpittaway** [comment]
Agreed, just run into this problem myself

**Comment by derekgroh** [comment]
This is actually an error with the .kitchen.yml file and not the code base itself. This cookbook has not been reworked for the changes to windows version 3.0.0. 

That being said, I see no reason to not include this change, as it is a good catch all moving forward.
```
provisioner:
  require_Chef_omnibus: 12.21.1
```
will fix the issue

**Comment by joenguyen** [comment]
FWIW, I'm still seeing that issue with `12.21.1`
```
             ================================================================================
             Error executing action `install` on resource 'windows_feature_dism[DirectoryServices-AdministrativeCenter]'
             ================================================================================

             Mixlib::ShellOut::ShellCommandFailed
             ------------------------------------
             Expected process to exit with [0, 42, 127, 3010], but received '50'
             ---- Begin output of C:\Windows\system32\dism.exe /online /enable-feature /featurename:DirectoryServices-AdministrativeCenter /norestart   ----
             STDOUT: Deployment Image Servicing and Management tool

             Version: 6.3.9600.17031



             Image Version: 6.3.9600.17031



             Enabling feature(s)



             Error: 50



             The operation is complete but DirectoryServices-AdministrativeCenter feature was not enabled.

             A required parent feature may not be enabled. You can use the /enable-feature /all option to automatically enable each parent feature from the following list. If the parent feature(s) are already enabled, refer to the log file for further diagnostics.

             ActiveDirectory-PowerShell



             The DISM log file can be found at C:\Windows\Logs\DISM\dism.log
             STDERR:
             ---- End output of C:\Windows\system32\dism.exe /online /enable-feature /featurename:DirectoryServices-AdministrativeCenter /norestart   ----
             Ran C:\Windows\system32\dism.exe /online /enable-feature /featurename:DirectoryServices-AdministrativeCenter /norestart   returned 50

             Resource Declaration:
             ---------------------
             # In C:/Users/vagrant/AppData/Local/Temp/kitchen/cache/cookbooks/windows/resources/feature.rb

       60:       windows_feature_dism new_resource.name do
       61:         action desired_action
       62:         feature_name new_resource.feature_name
       63:         source new_resource.source if new_resource.source
       64:         all new_resource.all
       65:       end
       66:     when :windows_feature_servermanagercmd

             Compiled Resource:
             ------------------
             # Declared in C:/Users/vagrant/AppData/Local/Temp/kitchen/cache/cookbooks/windows/resources/feature.rb:60:in `run_default_provider'

             windows_feature_dism("DirectoryServices-AdministrativeCenter") do
        action [:install]
        retries 0
        retry_delay 2
        default_guard_interpreter :default
        declared_type :windows_feature_dism
        cookbook_name "windows_ad"
        feature_name "DirectoryServices-AdministrativeCenter"
        all false
             end

             System Info:
             ------------
             chef_version=12.21.1
             platform=windows
             platform_version=6.3.9600
             ruby=ruby 2.3.1p112 (2016-04-26 revision 54768) [x64-mingw32]
             program_name=C:/opscode/chef/bin/chef-client
             executable=C:/opscode/chef/bin/chef-client

       [2017-07-26T08:34:55-07:00] INFO: Running queued delayed notifications before re-raising exception

           ================================================================================
           Error executing action `install` on resource 'windows_feature[DirectoryServices-AdministrativeCenter]'
           ================================================================================

           Mixlib::ShellOut::ShellCommandFailed
           ------------------------------------
           windows_feature_dism[DirectoryServices-AdministrativeCenter] (C:/Users/vagrant/AppData/Local/Temp/kitchen/cache/cookbooks/windows/resources/feature.rb line 60) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0, 42, 127, 3010], but received '50'
           ---- Begin output of C:\Windows\system32\dism.exe /online /enable-feature /featurename:DirectoryServices-AdministrativeCenter /norestart   ----
           STDOUT: Deployment Image Servicing and Management tool

           Version: 6.3.9600.17031



           Image Version: 6.3.9600.17031



           Enabling feature(s)



           Error: 50



           The operation is complete but DirectoryServices-AdministrativeCenter feature was not enabled.

           A required parent feature may not be enabled. You can use the /enable-feature /all option to automatically enable each parent feature from the following list. If the parent feature(s) are already enabled, refer to the log file for further diagnostics.

           ActiveDirectory-PowerShell



           The DISM log file can be found at C:\Windows\Logs\DISM\dism.log
           STDERR:
           ---- End output of C:\Windows\system32\dism.exe /online /enable-feature /featurename:DirectoryServices-AdministrativeCenter /norestart   ----
           Ran C:\Windows\system32\dism.exe /online /enable-feature /featurename:DirectoryServices-AdministrativeCenter /norestart   returned 50

           Resource Declaration:
           ---------------------
           # In C:/Users/vagrant/AppData/Local/Temp/kitchen/cache/cookbooks/windows_ad/recipes/default.rb

            39:     windows_feature feature do
            40:       action :install
            41:     end
            42:   end

           Compiled Resource:
           ------------------
           # Declared in C:/Users/vagrant/AppData/Local/Temp/kitchen/cache/cookbooks/windows_ad/recipes/default.rb:39:in `block in from_file'

           windows_feature("DirectoryServices-AdministrativeCenter") do
             action [:install]
             retries 0
             retry_delay 2
             default_guard_interpreter :default
             declared_type :windows_feature
             cookbook_name "windows_ad"
             recipe_name "default"
             feature_name "DirectoryServices-AdministrativeCenter"
           end

           System Info:
           ------------
           chef_version=12.21.1
           platform=windows
           platform_version=6.3.9600
           ruby=ruby 2.3.1p112 (2016-04-26 revision 54768) [x64-mingw32]
           program_name=C:/opscode/chef/bin/chef-client
           executable=C:/opscode/chef/bin/chef-client
```

**Comment by derekgroh** [comment]
What OS version?

**Comment by derekgroh** [comment]
@joenguyen ok I'll take a look.

**Comment by joenguyen** [comment]
Windows 2012r2

**Comment by derekgroh** [comment]
@joenguyen Change has been made and released as version 0.6.1 on supermaket. Please let me know if you have any other issues.
